specs/interop/derivation.md

```diff
@@ -87,17 +91,185 @@ The interop network upgrade timestamp defines the timestamp at which all functio
 the consensus rules for an OP Stack based network. On the interop network upgrade block, a set of deposit transaction
 based upgrade transactions are deterministically generated by the derivation pipeline in the following order:
 
-- L1 Attributes Transaction calling `setL1BlockValuesIsthmus`
-- User deposits from L1
-- Network Upgrade Transactions
-  - CrossL2Inbox deployment
-  - L2ToL2CrossDomainMessenger deployment
-  - OptimismMintableERC20Factory deployment
-  - Update CrossL2Inbox Proxy ERC-1967 Implementation Slot
-  - Update L2ToL2CrossDomainMessenger Proxy ERC-1967 Implementation Slot
-  - Update the OptimismMintableERC20Factory Proxy ERC-1967 Implementation Slot
-
-The execution payload MUST set `noTxPool` to `true` for this block.
+The execution payload MUST set `noTxPool` to `true` for this block, excluding user transactions from the block.
+
+This upgrade requires `3_320_000` in gas. In order to upgrade successfully the following invariant MUST be maintained:
+
+```plaintext
+MaxResourceLimit + SystemTxMaxGas + UpgradeGasUsage ≤ L2GasLimit
```

Comment:
Interesting, newbie question:
Just to make sure how do we deal with the `Deposit transaction`? 
My attacking scenario would be: 
Is a DoS possible to make a massive deposit transaction on L1, and break the invariants higher since the deposit transaction seems to not be included here? Thus to be bigger than the `L2GasLimit` and make the upgrade fail? 
Or the deposit transaction are excluded here somehow? 
because for me: "The execution payload MUST set `noTxPool` to `true`" should only affect L2 geth transaction but not the deposit one.

Reply:
The sum of gas used by deposits cannot be larger than the L2 block gas limit. The L1 contracts don't know when network upgrade transactions (issued as deposits) are going to be inserted. We rely on L1 contracts to prevent deposits from using too much L2 gas. There are protocol solutions to fix this (issue network upgrade txs over multiple blocks) but the most simple is to say that the L2 block gas limit must be larger than a certain amount for the upgrade to happen successfully. As long as that is true, then there is no way to dos. Its up to the chain operator to ensure that the gas limit is high enough, and its not an unreasonable gas limit

Reply:
The sum of gas used by deposits cannot be larger than the L2 block gas limit. The L1 contracts don't know when network upgrade transactions (issued as deposits) are going to be inserted. We rely on L1 contracts to prevent deposits from using too much L2 gas. There are protocol solutions to fix this (issue network upgrade txs over multiple blocks) but the most simple is to say that the L2 block gas limit must be larger than a certain amount for the upgrade to happen successfully. As long as that is true, then there is no way to dos. Its up to the chain operator to ensure that the gas limit is high enough, and its not an unreasonable gas limit

Reply:
The sum of gas used by deposits cannot be larger than the L2 block gas limit. The L1 contracts don't know when network upgrade transactions (issued as deposits) are going to be inserted. We rely on L1 contracts to prevent deposits from using too much L2 gas. There are protocol solutions to fix this (issue network upgrade txs over multiple blocks) but the most simple is to say that the L2 block gas limit must be larger than a certain amount for the upgrade to happen successfully. As long as that is true, then there is no way to dos. Its up to the chain operator to ensure that the gas limit is high enough, and its not an unreasonable gas limit